core/network/hostport.go

```diff
@@ -77,10 +96,23 @@ func (hps HostPorts) PrioritizedForScope(getMatcher ScopeMatchFunc) []string {
 	return out
 }
 
+// CanonicalURL returns a URL value for the input HostPort,
+// that includes the the provided scheme.
+func CanonicalURL(a HostPort, scheme string) url.URL {
+	hostPort := net.JoinHostPort(a.Host(), strconv.Itoa(a.Port()))
+	u := url.URL{
+		Scheme: scheme,
+		Host:   hostPort,
+		Path:   a.AddressPath(),
+	}
+	return u
+}
+
 // DialAddress returns a string value for the input HostPort,
 // suitable for passing as an argument to net.Dial.
 func DialAddress(a HostPort) string {
-	return net.JoinHostPort(a.Host(), strconv.Itoa(a.Port()))
+	hostPort := net.JoinHostPort(a.Host(), strconv.Itoa(a.Port()))
```

Comment:
It's likely inlined anyway, but why semantically add this assignment?

Reply:
This was a leftover from a previous change that I eventually reverted. Will revert this.